src/corehost/cli/deps_resolver.cpp

```diff
@@ -119,6 +122,197 @@ void deps_resolver_t::get_dir_assemblies(
+    }
+
+    append_path(&path, max_str.c_str());
+
+    return entry.to_rel_path(path, candidate);
+}
+
+void deps_resolver_t::setup_probe_config(
+    const corehost_init_t* init,
+    const runtime_config_t& config,
+    const arguments_t& args)
+{
+    if (pal::directory_exists(args.dotnet_extensions))
+    {
+        pal::string_t ext_ni = args.dotnet_extensions;
+        append_path(&ext_ni, get_arch());
+        if (pal::directory_exists(ext_ni))
+        {
+            // Servicing NI probe.
+            m_probes.push_back(probe_config_t::svc_ni(ext_ni, config.get_fx_roll_fwd()));
+        }
+
+        // Servicing normal probe.
+        m_probes.push_back(probe_config_t::svc(args.dotnet_extensions, config.get_fx_roll_fwd()));
+    }
+
+    if (pal::directory_exists(args.dotnet_packages_cache))
+    {
+        pal::string_t ni_packages_cache = args.dotnet_packages_cache;
+        append_path(&ni_packages_cache, get_arch());
+        if (pal::directory_exists(ni_packages_cache))
+        {
+            // Packages cache NI probe
+            m_probes.push_back(probe_config_t::cache_ni(ni_packages_cache));
+        }
+
+        // Packages cache probe
+        m_probes.push_back(probe_config_t::cache(args.dotnet_packages_cache));
+    }
+
+    if (pal::directory_exists(m_fx_dir))
+    {
+        // FX probe
+        m_probes.push_back(probe_config_t::fx(m_fx_dir, m_fx_deps.get()));
+    }
+
+    for (const auto& probe : m_additional_probes)
+    {
+        // Additional paths
+        assert(pal::directory_exists(probe));
+        // FIXME: When removing additional probes add back config get fx roll fwd.
+        m_probes.push_back(probe_config_t::additional(probe,  false /* config.get_fx_roll_fwd() */));
+    }
+
+    if (trace::is_enabled())
+    {
+        trace::verbose(_X("-- Listing probe configurations..."));
+        for (const auto& pc : m_probes)
+        {
+            pc.print();
+        }
+    }
+}
+
+void deps_resolver_t::setup_additional_probes(const std::unordered_set<pal::string_t>& probe_paths)
+{
+    m_additional_probes.assign(probe_paths.begin(), probe_paths.end());
+
+    for (auto iter = m_additional_probes.begin(); iter != m_additional_probes.end(); )
+    {
+        if (pal::directory_exists(*iter))
+        {
+            ++iter;
+        }
+        else
+        {
+            iter = m_additional_probes.erase(iter);
+        }
+    }
+    // FIXME: Remove nuget support with runtimeconfig.dev.json
+    // and setup roll forward to come from the config in setup_probe_config
+    if (m_additional_probes.empty())
+    {
+        pal::string_t probe_dir;
+        (void)pal::get_default_packages_directory(&probe_dir);
+        if (pal::directory_exists(probe_dir))
+        {
+            m_additional_probes.push_back(probe_dir);
+        }
+    }
+}
+
+bool deps_resolver_t::probe_entry_in_configs(const deps_entry_t& entry, pal::string_t* candidate)
+{
+    candidate->clear();
+    for (const auto& config : m_probes)
```

Comment:
How is the order maintained for probing? That is, for each package, how do we ensure the following remains true:

dotnetextension
hosting_optimization_Cache
FX
additionalprobingpaths at command line
additionalprobingpaths in runtimeconfig.dev.json